scripts/entrypoint.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2021 Bosch.IO GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# License-Filename: LICENSE
+
+set -euo pipefail
+
+if [[ -f /root/.ssh/id_rsa ]]; then
```

Comment:
I'm not sure this is the right logic to determine whether run the agent *inside* the container, or whether to rely on it running on the *host*. I probably would have expected a condition that checks for the `SSH_AUTH_SOCK` environment variable to be set. If yes, the agent is running on the host and it gets forwarded to the container. If no, we need to run the agent ourselves.